CMakeLists.txt

```diff
@@ -0,0 +1,108 @@
+cmake_minimum_required(VERSION 3.2.2)
+# partially inspired by https://rix0r.nl/blog/2015/08/13/cmake-guide/
+
+set(LIBRARY_NAME date)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+project(${LIBRARY_NAME} LANGUAGES CXX)
+
+set(MAJOR_VERSION 1)
+set(MINOR_VERSION 0)
+set(PATCH_VERSION 0)
+
+# Determine whether this is a standalone project or included by other projects
+set(${LIBRARY_NAME}_STANDALONE_PROJECT OFF)
+if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+  set(${LIBRARY_NAME}_STANDALONE_PROJECT ON)
+endif ()
+
+# Use GNUInstallDirs to install libraries into correct locations on all platforms.
+include(GNUInstallDirs)
+
+
+################################################################################
+# Add the library
+
+add_library(${LIBRARY_NAME} 
+    src/tz.cpp)
+
+target_include_directories(${LIBRARY_NAME} PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> # for headers when building
+    $<INSTALL_INTERFACE:include/${LIBRARY_NAME}> # for client in install mode
+    PRIVATE src
+    )
+
+install(TARGETS ${LIBRARY_NAME} EXPORT ${LIBRARY_NAME}  
+    RUNTIME  DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE  DESTINATION ${CMAKE_INSTALL_LIBDIR})  
+
+install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${LIBRARY_NAME})
+
+# This makes the project importable from the install directory
+# Put config file in per-project dir (name MUST match), can also
+# just go into 'cmake'.
+install(EXPORT ${LIBRARY_NAME} DESTINATION share/${LIBRARY_NAME}/cmake)
+
+# This makes the project importable from the build directory
+export(TARGETS ${LIBRARY_NAME} FILE ${LIBRARY_NAME}.cmake)
+
+###############################################################################
+# Packaging configuration
+if (${LIBRARY_NAME}_STANDALONE_PROJECT)
```

Comment:
Again this seems Debian specific and RPATH sets off a red flag for me (though maybe that's a problem only on Debian). Surely there's a better way to handle other distros as well?

Reply:
CPack allows a per target configuration. What you see here is indeed the Debian specific part, which is ignored on other platforms. For RPM we have to add a similar section. On windows everything goes to the standard places (admittedly not tested yet).

With regard to RPATH: This configuration installs into `/opt/date-VERSION` and I tried all variants with and without RPATH in [continuous-integration-demo](https://github.com/daixtrose/continuous-integration-demo) and came to the conclusion that a relative RPATH search, i.e. an executable that prefers its **own** library is exactly what you need in case you have an executable.

I kept this entry here because I was thinking about an option to package some of the test executables into the distro. Not sure though ...